src/openms/source/TRANSFORMATIONS/RAW2PEAK/PeakPickerCWT.cpp

```diff
@@ -1222,7 +1222,7 @@ namespace OpenMS
              && (shape.getFWHM() >= fwhm_bound_)
              && (shape.getFWHM() <= fwhm_upper_bound))
           {
-            shape.signal_to_noise = sne.getSignalToNoise(area.max);
+            shape.signal_to_noise = sne.getSignalToNoise((Size)distance(raw_peak_array.begin(),area.max));
             peak_shapes.push_back(shape);
```

Comment:
Is it possible to use `emplace_back ` instead?

Reply:
It is possible.